### DISCUSSION TITLE
Pull gives `remote missing object` error for LFS file, but I can download it manually from repository website.

### DISCUSSION THREAD

**Discussion by IsaacBrown-BPA** [community/community#39669] [New to GitHub]
I'm trying to collaborate with a coworker and he just pushed a commit adding tracking for a few more files, including a new LFS file. When I tried to pull this commit, I got an error:

`Error downloading object: <TheLFSFile> (<fileHash>): Smudge error: Error downloading <TheLFSFile> (<fileHash>): error transferring "<fileHash>": [0] remote missing object <fileHash>`

When I come here to GitHub and go to the code page for the repository and click on the file in question, I can download it directly without issue, so it's clearly on the server. If I just manually download this file and try to pull the commit, git complains that it would overwrite untracked files, so I'm kind of stuck unable to pull this commit.

I know git added some path security, and we had to solve a similar LFS issue when using a bare repository on a network drive as a remote. We had to add the network path to the global config as a trusted one like so:

`git config --global --add safe.directory '<NetworkPathToRepo>'` 

Is there something similar I have to do to add a github repository path as a trusted LFS path?
In the short-term, is there a way to force the pull to skip LFS files (I found tons of posts about skipping them when cloning, but nothing about skipping them when pulling)?

EDIT: We're both on Windows, he uses TortoiseGit and I use SourceTree and CLI. For S&Gs I tried GitHub Desktop and got the same result.

**Comment by IsaacBrown-BPA** [comment]
The reset of the LFS log after the bit already pasted in the first post:

```

github.com/git-lfs/git-lfs/v3/errors.New
        github.com/git-lfs/git-lfs/v3/errors/errors.go:62
github.com/git-lfs/git-lfs/v3/tq.(*customAdapter).DoTransfer
        github.com/git-lfs/git-lfs/v3/tq/custom.go:311
github.com/git-lfs/git-lfs/v3/tq.(*adapterBase).worker
        github.com/git-lfs/git-lfs/v3/tq/adapterbase.go:183
runtime.goexit
        runtime/asm_amd64.s:1571
Error downloading <TheLFSFile> (<fileHash>)
github.com/git-lfs/git-lfs/v3/errors.newWrappedError
        github.com/git-lfs/git-lfs/v3/errors/types.go:229
github.com/git-lfs/git-lfs/v3/errors.Wrapf
        github.com/git-lfs/git-lfs/v3/errors/errors.go:85
github.com/git-lfs/git-lfs/v3/lfs.(*GitFilter).downloadFile
        github.com/git-lfs/git-lfs/v3/lfs/gitfilter_smudge.go:120
github.com/git-lfs/git-lfs/v3/lfs.(*GitFilter).Smudge
        github.com/git-lfs/git-lfs/v3/lfs/gitfilter_smudge.go:79
github.com/git-lfs/git-lfs/v3/commands.smudge
        github.com/git-lfs/git-lfs/v3/commands/command_smudge.go:128
github.com/git-lfs/git-lfs/v3/commands.filterCommand
        github.com/git-lfs/git-lfs/v3/commands/command_filter_process.go:123
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v0.0.3/command.go:766
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v0.0.3/command.go:852
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v0.0.3/command.go:800
github.com/git-lfs/git-lfs/v3/commands.Run
        github.com/git-lfs/git-lfs/v3/commands/run.go:107
main.main
        github.com/git-lfs/git-lfs/v3/git-lfs.go:34
runtime.main
        runtime/proc.go:250
runtime.goexit
        runtime/asm_amd64.s:1571
Smudge error
github.com/git-lfs/git-lfs/v3/errors.newWrappedError
        github.com/git-lfs/git-lfs/v3/errors/types.go:229
github.com/git-lfs/git-lfs/v3/errors.NewSmudgeError
        github.com/git-lfs/git-lfs/v3/errors/types.go:315
github.com/git-lfs/git-lfs/v3/lfs.(*GitFilter).Smudge
        github.com/git-lfs/git-lfs/v3/lfs/gitfilter_smudge.go:88
github.com/git-lfs/git-lfs/v3/commands.smudge
        github.com/git-lfs/git-lfs/v3/commands/command_smudge.go:128
github.com/git-lfs/git-lfs/v3/commands.filterCommand
        github.com/git-lfs/git-lfs/v3/commands/command_filter_process.go:123
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v0.0.3/command.go:766
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v0.0.3/command.go:852
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v0.0.3/command.go:800
github.com/git-lfs/git-lfs/v3/commands.Run
        github.com/git-lfs/git-lfs/v3/commands/run.go:107
main.main
        github.com/git-lfs/git-lfs/v3/git-lfs.go:34
runtime.main
        runtime/proc.go:250
runtime.goexit
        runtime/asm_amd64.s:1571

Current time in UTC:
2022-11-21 13:55:15

Environment:
LocalWorkingDir=<MyRepoDirectory>
LocalGitDir=<MyRepoDirectory>\.git
LocalGitStorageDir=<MyRepoDirectory>\.git
LocalMediaDir=<MyRepoDirectory>\.git\lfs\objects
LocalReferenceDirs=
TempDir=<MyRepoDirectory>\.git\lfs\tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=<MyRepoDirectory>\.git\lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,bitbucket-media-api,lfs-standalone-file,ssh
UploadTransfers=basic,bitbucket-media-api,lfs-standalone-file,ssh
GIT_CONFIG_PARAMETERS='color.branch'='false' 'color.diff'='false' 'color.status'='false' 'diff.mnemonicprefix'='false' 'core.quotepath'='false' 'lfs.customtransfer.bitbucket-media-api.path'='git-lfs-bitbucket-media-api'
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core
GIT_OPTIONAL_LOCKS=0
GIT_PAGER=cat
GIT_PREFIX=
GIT_REFLOG_ACTION=pull --no-commit GitHub main

```

**Comment by david-engelmann** [comment]
have you tried running the `git lfs migrate` command?

**Reply by IsaacBrown-BPA** [reply]
No...

It's my understanding that `git lfs migrate` is to convert files to or from LFS. Are you suggesting we just don't use LFS to bypass this issue by migrating the file out of LFS? In that case, we'd probably just make a new repository instead of fooling around with lfs migration (EDIT: Because there are currently several LFS tracked files in the repo and will be several more as we go along with the project).

Or does `git lfs migrate` have other options you're alluding to?

**Reply by david-engelmann** [reply]
[Check out this stackoverflow](https://stackoverflow.com/questions/54670485/what-can-cause-a-git-lfs-object-to-go-missing), it will link to this [discussion on the topic](https://stackoverflow.com/questions/35518688/git-lfs-refused-to-track-my-large-files-properly-until-i-did-the-following/35518689#35518689). It appears to occur with SourceTree users occasionally

**Reply by IsaacBrown-BPA** [reply]
We checked on my colleague's end and the file shows up as tracked doing `git lfs ls-files` and the file is there on GitHub, so their push worked fine. Of course it isn't on my end yet since I haven't successfully pull the commit. Not sure how `git lfs migrate` is going to change things

**Reply by david-engelmann** [reply]
[Check out the docs on it](https://github.com/git-lfs/git-lfs/blob/v2.2.0/docs/man/git-lfs-migrate.1.ronn)
